src/Build.UnitTests/Scanner_Tests.cs

```diff
@@ -104,16 +105,27 @@ public void IllFormedProperty()
         /// <summary>
         /// Tests the space errors case
         /// </summary>
-        [Fact]
-        public void SpaceProperty()
+        [Theory]
+        [InlineData("$(x )")]
+        [InlineData("$( x)")]
+        public void SpaceProperty(string pattern)
         {
-            Scanner lexer = new Scanner("$(x )", ParserOptions.AllowProperties);
+            Scanner lexer = new Scanner(pattern, ParserOptions.AllowProperties);
             AdvanceToScannerError(lexer);
             Assert.Equal("IllFormedPropertySpaceInCondition", lexer.GetErrorResource());
+        }
 
-            lexer = new Scanner("$( x)", ParserOptions.AllowProperties);
+        /// <summary>
+        /// Tests the space not next to end so no errors case
+        /// </summary>
+        [Theory]
+        [InlineData("$(x x)")]
+        [InlineData("$(y d f  ( hi ) sx)")]
```

Comment:
These are examples of things we wish we could catch but we can't with current limitations. Can you instead use examples we want to work like
```suggestion
        [InlineData("$(x.StartsWith( 'y' ))")]
        [InlineData("$(x.StartsWith ('y'))")]
```
?

Reply:
Actually, looking at this, I think even starts/ends with space is too aggressive. `$( x.StartsWith ( y ) )` is a bit ugly but works just fine today. Maybe it has to be "has a space _and_ is otherwise only identifier characters"? That would correctly reject `$(x x)` from your examples here.